timemachine/fe/free_energy.py

```diff
@@ -1165,6 +1165,17 @@ def compute_dense():
 
     U_kl = compute_sparse(max_delta_states) if max_delta_states is not None else compute_dense()
 
+    # Verify that all of the diagonal energies are finite, else a state has blown up
+    diag_idxs = np.diag_indices_from(U_kl)
+    assert np.all(np.isfinite(U_kl[diag_idxs])), "A state is no longer valid"
+    assert np.all(np.abs(U_kl[diag_idxs]) < 1e9), "Energies larger in magnitude than tolerated"
+    if np.any(np.isnan(U_kl)):
+        warn(
+            "Encountered NaNs in U_kl matrix. Replacing each instance with inf prior to HREX swaps",
+            IndeterminateEnergyWarning,
+        )
+        U_kl = np.where(np.isnan(U_kl), np.inf, U_kl)
+
```

Comment:
nit: would prefer to put this code in a separate function (`sanitize_potential_matrix_for_hrex` or similar?) and update the usage in `run_sims_hrex` to something like

```python
U_kl_raw = compute_potential_matrix(...)
U_kl = sanitize_potential_matrix_for_hrex(U_kl_raw, abs_energy_threshold=1e9)
```

to avoid entangling `compute_potential_matrix` with HREX.

Reply:
(The assertion that the energies have not blown up seems useful to have independently of HREX.)